akka-stream-tests/src/test/scala/akka/stream/scaladsl/SourceSpec.scala

```diff
@@ -436,4 +437,18 @@ class SourceSpec extends StreamSpec with DefaultTimeout {
       a[RuntimeException] shouldBe thrownBy(matValPoweredSource.preMaterialize())
     }
   }
+
+  "Source.futureSource" must {
+
+    "not cancel substream twice" in assertAllStagesStopped {
+      val result = Source
+        .futureSource(akka.pattern.after(2.seconds)(Future.successful(Source(1 to 2))))
+        .merge(Source(3 to 4))
+        .take(1)
+        .runWith(Sink.ignore)
+
+      Await.ready(result, 4.seconds)
+      result.value shouldBe Some(util.Success(Done))
```

Comment:
```
Await.result(result, 4.seconds) shouldBe Done
```

?

Reply:
Was playing around, will revert that to Await.result

Reply:
Updated the commit, should be ready to merge